app/main.cpp

```diff
@@ -54,26 +52,28 @@ void installTranslator( PencilApplication& app )
     qDebug() << "Install translation = " << b;
 }
 
-int handleArguments( MainWindow2 & mainWindow )
+int handleArguments( PencilApplication& app )
 {
+    QTextStream out( stdout );
+    QTextStream err( stderr );
```

Comment:
Do you think qWarning() and qCritical() would be better than QTextStreams? They exist for the specific function of printing warnings and errors, and can easily be controlled with defines, however we would have to use something something like qCritical().noQuote() or convert the QString to a const char * on each output to prevent it from automatically quoting the translated QStrings, which is quite ugly.

Reply:
To me it seems that all of these q*() functions are mainly for logging, not for cli output that is meant to be seen by the user. For example, take this from the documentation of qWarning() and qCritical():

> Under Windows, the message is sent to the debugger. On QNX the message is sent to slogger2.

Imo cli output is not something that should be sent to debuggers or system logging daemons under any circumstances. And ftr, QCommandLineParser also prints directly to stdout/stderr (using fputs) on all systems except WinRT and possibly Windows.

Reply:
> Imo cli output is not something that should be sent to debuggers or system logging daemons under any circumstances

I agree. I did not realize it did that for non-qDebug() functions. We can leave it as is then.